repository/src/main/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceConfig.java

```diff
@@ -103,9 +108,34 @@ public int getConnectionPoolSize()
         return connectionPoolSize;
     }
 
+    public void setIssuerUrl(String issuerUrl)
+    {
+        this.issuerUrl = issuerUrl;
+    }
+
+    public String getAudience()
+    {
+        return audience;
+    }
+
+    public void setAudience(String audience)
+    {
+        this.audience = audience;
+    }
+
     public String getAuthServerUrl()
     {
-        return authServerUrl;
+        if(StringUtils.isNotBlank(realm) && StringUtils.isNotBlank(authServerUrl))
+        {
+            return UriComponentsBuilder.fromUriString(authServerUrl)
+                .pathSegment(REALMS, getRealm())
+                .build()
+                .toString();
+        }
+        else
+        {
+            return authServerUrl;
+        }

```

Comment:
```suggestion

            return authServerUrl;
```

No need for the `else` clause. Just return `authServerUrl` immediately.

Reply:
Done. I replaced the whole logic with that Optional flow. There was a need to change the last line.